### PR TITLE
Full Featured Using and Omitting

### PR DESCRIPTION
**Pull request by sheavalentine-mdsol**
Added the rest of the agent relevent methods
Exposed the methods in Representor and Transition objects
Lots of tests

### CODE REVIEW

File: lib/farscape/transition.rb [hunk 1]
```diff
@@ -28,6 +29,12 @@ def invoke(args={})
     end
 
     def method_missing(meth, *args, &block)
+      if [:omitting, :using].include?(meth)
```
Comment (HonoreDB) [inline lib/farscape/transition.rb]: I'd rather not define statically-named methods using method_missing, it breaks stuff like `respond_to`, `methods`, and doc generation. If you want to DRY these methods, you can either use a delegation library like [active_support's](http://guides.rubyonrails.org/active_support_core_extensions.html#method-delegation) or use `define_method`.

**Comment by HonoreDB** [comment]
This looks great, one comment on implementation but +1 to the overall design.

File: lib/farscape/transition.rb (hunk 1 above)
Comment (sheavalentine-mdsol) [inline lib/farscape/transition.rb]: Good, I didn't like it either.  I'll look into define_method.

**Comment by sheavalentine-mdsol** [comment]
Travis is being weird.

**Comment by sheavalentine-mdsol** [comment]
@HonoreDB  - Any other comments?

**Pull request merged**